How to create switch statement inside rendering method in Next.js? (#15)
XAP-Software/blog.xap.group: _posts/2022-05-11-switch_statement_next_js.md

---
layout: post
title: How to create switch statement inside rendering method in Next.js?
categories: [Next.js]
author_github: negativename
author: Vladislav Kobenko
---

> In case you want to create switch statement in rendering method in `Next.js` like this:

Reviewer: ```suggestion
In case you want to create a switch statement in rendering method in `Next.js` similar to the one below you can use on of the methods described in this post.
```

```jsx
return (
        <div>
            <div>
                // ...some info...
            </div>

           { switch(...) {} }

            <div>
                // ...some info...
            </div>
        </div>
    );
```

> You can use one of the following methods.

Reviewer: ```suggestion
```

## Creating self-invoking function

> The first method is to use self-invoking function:

Reviewer: ```suggestion
The first method is to use a self-invoking function:
```

```jsx
return (
        <div>
            <div>
                // ...some info...
            </div>

           {(() => {
               switch(param) {
                    case 'A':
                       return (<p>Some info</p>)
                    case 'B':
                       return (<span>Another info</span>)
               } 
           })()}

            <div>
                // ...some info...
            </div>
        </div>
    );
```

> Now your `switch` statement will invoke like `JS` expression. 

Reviewer: ```suggestion
Now your `switch` statement will invoke the `JS` expression. 
```

## Creating method/component outside of rendering method

> An another way is to create method or component with `switch` statement outside of rendering method. For example:

Reviewer: ```suggestion
Another way is to create a method or component with the `switch` statement outside of the rendering method. For example:
```

```jsx
// write this code outside of rendering method

const test = (param) => {
    switch (param){
        case 'A':
            return(
                <p>Some info</p>
            )
        case 'B':
            return(
                <span>Another info</span>
            ) 
    }
}
```

> And in the rendering method you should to invoke your `test` method:

Reviewer: ```suggestion
Then in the rendering method you can invoke your `test` method:
```
